cpp/diffraction/StaticStructureFactorDebye.cc

```diff
@@ -15,9 +15,42 @@
 
 namespace freud { namespace diffraction {
 
+namespace {
```

Comment:
Why the empty namespace?

Reply:
> Unnamed namespaces are typically used when you have a lot of content that you want to ensure stays local to a given file [...]

https://www.learncpp.com/cpp-tutorial/unnamed-and-inline-namespaces/

These helper functions shouldn't be a part of the `freud::diffraction` public namespace. We only want to use them in this translation unit. Note: anonymous namespaces should never be used in headers (`*.h`), only implementation files (`*.cc`).

C++ Core Guidelines has advice on these topics as well:
- [SF.22: Use an unnamed (anonymous) namespace for all internal/non-exported entities](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#sf22-use-an-unnamed-anonymous-namespace-for-all-internalnon-exported-entities)
- [SF.21: Don’t use an unnamed (anonymous) namespace in a header](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rs-unnamed)